[samsungmobile] Add Galaxy Xcover 4

adriens opened the pull request:
- [Released in April 2017](https://en.wikipedia.org/wiki/Samsung_Galaxy_Xcover_4)
- [Official Product  homepage](https://www.samsung.com/ch_fr/business/smartphones/galaxy-xcover-4-g390f/sm-g390fzkaaut/)
- [Security maintenance release schedule](https://docs.samsungknox.com/admin/knox-suite/kbas/kba-367-enterprise-edition-release-schedule.htm)
- [Samsung Security Updates](https://security.samsungmobile.com/workScope.smsb)
- Current OS :  [Android 9 "Pie"](https://en.wikipedia.org/wiki/Android_Pie)
- [Android Pie](https://en.wikipedia.org/wiki/Android_Pie)  final release : **January 4, 2022**

![image](https://user-images.githubusercontent.com/5235127/216461219-89f3d2aa-52ec-4189-ac07-255a53fc6084.png)

adriens committed:
Add Galaxy Xcover 4

adriens committed:
Update samsungmobile.md

adriens committed:
No security update since december 2019

adriens reviewed (commented):
As of 
![image](https://user-images.githubusercontent.com/5235127/216461556-2bde38ed-a64a-41c9-ac7c-d6443d931f49.png)

adriens committed:
Update samsungmobile.md

BiNZGi commented on products/samsungmobile.md:
```diff
@@ -153,6 +153,11 @@ releases:
 
+-   releaseCycle: "Galaxy Xcover 4"
+    support: false
+    eol: 2020-11-19
```
I don't find this date in your mentioned links. I would change it to `2019-12-01` as last patch date in your screenshot.

adriens committed:
Xcover 4 EOL

cf https://github.com/endoflife-date/endoflife.date/pull/2439#discussion_r1099875205
